scripts/lint-bib.sh

```diff
@@ -6,6 +6,17 @@ set -euo pipefail
 IFS=$'\n\t'
 
 set -x
+
+# Test if there are keys with non-ASCII characters.
+bibtool --pass.comments=on -- 'select{$key "[^-.:A-Za-z0-9_]+"}' \
+  docs/references.bib -o docs/non-ascii.bib
```

Comment:
should `docs/non-ascii.bib` be added to the `.gitignore` file?

Reply:
That file is only temporarily during lint process. Should I delete that file after the process finished?

Reply:
I don't think it needs to be added to `.gitignore` file, as the `docs/references.bib.old` (a few lines below) is not in `.gitignore` either.

Reply:
Not sure what the error below is about permissions means...

Maybe, would it be more robust to use a variable instead of a temp file? I think that would be with `invalid_keys=$(bibtool ...)` and then `if [[ -n "$invalid_keys" ]]; then`. Or is it desired that the file with the errors exists?

Reply:
Ah and, I think you'd need `rm -f docs/non-ascii.bib.tmp`. A blank `rm` might fail if the file does not exist

Reply:
Thank you! Now I use a variable instead of a temp file. But now we need to figure out that permission error before we see if it works...

Reply:
Now it should works, as the test case https://github.com/leanprover-community/mathlib4/actions/runs/13896105506/job/38876988444?pr=22757 reports errors correctly.